### PR TITLE
add flag to ignore requests scaling up or down

### PR DESCRIPTION
**Pull request by ganga1980**
#### Which component this PR applies to?

addon-resizer
#### What type of PR is this?
feature 

#### What this PR does / why we need it:
This PR exposes flag to ignore the scaling of the requests and only scale the limits. In large clusters, scaling requests will cause the scheduling issues for other workload pods.  This flag enables to scale only the limits and have the specified fixed requests as is.

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?
Yes

```release-note
Introduced --ignoreResourceRequests flag to ignore the scaling of the resource requests.
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by ganga1980** [comment]
Hi, @jbartosik, @wojtek-t, can you guys help on reviewing this PR?

**Comment by ganga1980** [comment]
Hi, @jbartosik , @wojtek-t , Appreciate your help on reviewing this PR as I would like to get this change merged soon.

**Comment by voelzmo** [comment]
I can't really comment on the functionality itself so here's just a comment regarding the "how it is done": given that we have already a setting for which values are to be controlled by the VPA ([`controlledValues`](https://github.com/kubernetes/autoscaler/blob/master/vertical-pod-autoscaler/pkg/apis/autoscaling.k8s.io/v1/types.go#L184) with currently possible values `RequestsOnly` and `RequestsAndLimits`), it would probably better fit in there as `LimitsOnly`.

**Comment by ganga1980** [comment]
Hi,  @jbartosik , can you please help on reviewing this PR?

**Comment by ganga1980** [comment]
Hi, @jbartosik, @wojtek-t , sorry to bug you guys on this. If you guys don't have cycles to review, can you please tag the folks who can help on reviewing this PR?

**Comment by jbartosik** [comment]
Hi, I don't have capacity to review this. I'm pretty sure Wojciech doesn't either.

I'll ask if someone wants to review this on the next [SIG meeting](https://github.com/kubernetes/community/blob/master/sig-autoscaling/README.md#meetings) on Monday. You can try also try asking for volunteers on Slack.

**Comment by shreyas-badiger** [comment]
@ganga1980 
The description in the first comment is not entirely clear. Can you please update the PR with the following? 
- Clear documentation of the flag and its requirement.
- Testing proof, unit tests
- PR details for the release notes

File: addon-resizer/README.md [hunk 1]
```diff
@@ -28,6 +28,7 @@ Usage of pod_nanny:
       --storage="MISSING": The base storage resource requirement.
       --threshold=0: A number between 0-100. The dependent's resources are rewritten when they deviate from expected by more than threshold.
       --use-metrics=false: Whether to use apiserver metrics to detect cluster size instead of the default method of listing node objects from the Kubernetes API.
+      --ignoreResourceRequests=false: ignoring scaling up or down of the resource requests if this flag is true.
```
Comment (mwielgus) [inline addon-resizer/README.md:31]: It is hard to tell what this flag actually does when specified. Expand the description and/or change the flag name (for example scaleResourceRequest with the default set to true would be much more intuitive).

File: addon-resizer/nanny/kubernetes_client_test.go [hunk 2]
```diff
@@ -97,7 +97,7 @@ func TestMergeResources(t *testing.T) {
 
 	for _, tc := range testCases {
 		t.Run(tc.name, func(t *testing.T) {
-			got := mergeResources(tc.current, tc.new)
+			got := mergeResources(tc.current, tc.new, false)
```
Comment (mwielgus) [inline addon-resizer/nanny/kubernetes_client_test.go:100]: This is a new functionality and thus requires some tests. Please add them.

File: addon-resizer/nanny/nanny_lib_test.go [hunk 3]
```diff
@@ -219,7 +219,7 @@ func TestShouldOverwriteResources(t *testing.T) {
 	}
 	for i, tc := range testCases {
 
-		gotOverwrite, gotOp := shouldOverwriteResources(tc.th, tc.x, tc.x, tc.y, tc.x)
+		gotOverwrite, gotOp := shouldOverwriteResources(tc.th, tc.x, tc.x, tc.y, tc.x, false)
```
Comment (mwielgus) [inline addon-resizer/nanny/nanny_lib_test.go:222]: Add tests with true.

**Review by mwielgus — CHANGES_REQUESTED** [review]

**Pull request closed without merging**